hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestKeyDeletingService.java

```diff
@@ -139,8 +143,22 @@ public void checkIfDeleteServiceWithFailingSCM()
     createAndDeleteKeys(keyManager, keyCount, 1);
     KeyDeletingService keyDeletingService =
         (KeyDeletingService) keyManager.getDeletingService();
-    Assert.assertEquals(
-        keyManager.getPendingDeletionKeys(Integer.MAX_VALUE).size(), keyCount);
+    GenericTestUtils.waitFor(
+        () -> {
+          try {
+            int numPendingDeletionKeys =
+                keyManager.getPendingDeletionKeys(Integer.MAX_VALUE).size();
+            if (numPendingDeletionKeys != keyCount) {
+              LOG.error("Expected {} keys to be pending deletion, but got {}",
```

Comment:
I don't think a small delay is an error here.

```suggestion
              LOG.info("Expected {} keys to be pending deletion, but got {}",
```

Reply:
Done ✅